ludwig/data/dataset/ray.py

```diff
@@ -280,7 +280,8 @@ def __len__(self):
         try:
             count = next(self.epoch_iter).count()
         except TypeError:
-            count = next(self.epoch_iter.iter_batches()).count()
+            # Sum over all batches when using a DatasetIterator
+            count = sum(map(lambda b: b.count(), self.epoch_iter.iter_batches()))
```

Comment:
Couple of quick questions:

1. When would we hit the TypeError? Is it in the case that self.epoch_iter is not a DatasetIterator object?
2. Does this end up doing a pass over the entire data and call count on each batch, then sum that total?

Reply:
1. Yes, that's for backwards compatibility. I'm leaving it in for now just in case, but I think 2.3+ should follow the DatasetIterator path.
2. Yes. `iter_batch` returns an iterator over the batches of one epoch. It batches by block if batch_size isn't provided, but we pass batch_size to the actual batcher in a separate iter_batches call.

Reply:
2. The cost is relative to dataset size and object store memory, but running time for a number of different dataset sizes seems to be roughly the same as the other methods of getting dataset size from Ray Data.